src/main/java/uk/gov/hmcts/ccd/domain/service/processor/GlobalSearchProcessorService.java

```diff
@@ -146,17 +194,42 @@ private SearchPartyValue populateSearchParty(
             }
         }
 
-        searchPartyValue.setAddressLine1(findValueInMap(searchPartyDefinition.getSearchPartyAddressLine1(), data));
-        searchPartyValue.setEmailAddress(findValueInMap(searchPartyDefinition.getSearchPartyEmailAddress(), data));
-        searchPartyValue.setPostCode(findValueInMap(searchPartyDefinition.getSearchPartyPostCode(), data));
-        searchPartyValue.setDateOfBirth(findValueInMap(searchPartyDefinition.getSearchPartyDob(), data));
-        searchPartyValue.setDateOfDeath(findValueInMap(searchPartyDefinition.getSearchPartyDod(), data));
+        searchPartyValue.setAddressLine1(findTextValueInMap(searchPartyDefinition.getSearchPartyAddressLine1(), data));
+        searchPartyValue.setEmailAddress(findTextValueInMap(searchPartyDefinition.getSearchPartyEmailAddress(), data));
+        searchPartyValue.setPostCode(findTextValueInMap(searchPartyDefinition.getSearchPartyPostCode(), data));
+        searchPartyValue.setDateOfBirth(findTextValueInMap(searchPartyDefinition.getSearchPartyDob(), data));
+        searchPartyValue.setDateOfDeath(findTextValueInMap(searchPartyDefinition.getSearchPartyDod(), data));
 
         return searchPartyValue;
     }
 
-    private String findValueInMap(String valueToFind, Map<String, JsonNode> mapToSearch) {
-        String returnValue = null;
+    private String findTextValueInMap(String valueToFind, Map<String, JsonNode> mapToSearch) {
+        JsonNode value = findValueInMap(valueToFind, mapToSearch);
+        return value != null ? value.textValue() : null;
+    }
+
+    private List<JsonNode> findCollectionValueInMap(uk.gov.hmcts.ccd.domain.model.definition.SearchParty searchParty,
+                                                    Map<String, JsonNode> mapToSearch) {
+        JsonNode valueInMap = findValueInMap(searchParty.getSearchPartyCollectionFieldName(), mapToSearch);
+
+        List<JsonNode> listOfNodes = new ArrayList<>();
+
+        if (valueInMap != null) {
+            if (!valueInMap.isArray()) {
+                log.warn("GlobalSearch: This is not a collection in "
+                    + "the SearchParty tab and it should be. CaseTypeId: {}", searchParty.getCaseTypeId());
```

Comment:
Fixed in [6c80060](https://github.com/hmcts/ccd-data-store-api/pull/1828/commits/6c8006042492882e7a6ce9e8fd26e7339af29e99)

I have added the field name to the log - is this okay?

Reply:
This has been refactored to use the path to the collection field rather than the data value (which we should not be logging) in 236be0d38370e1db2863ce1706db48deedba2823